etc/scripts/generate_benchmarks.bash

```diff
@@ -39,31 +38,39 @@ fi
 NAME_PREFIX=$1
 NUM_BENCHMARKS=$2
 
-echo "Creating Backup of original config"
-cp ${CONFIG} ${CONFIG}.bak
-echo "Setup RefBox config"
-sed -E -n -i '1h;1!H;${g;s/(mongodb:[[:space:]]*\n[[:space:]]*enable:[[:space:]]*)false/\1true/;p;}' ${CONFIG} &>/dev/null
-sed -E -n -i '1h;1!H;${g;s/(time-sync:[[:space:]]*\n[[:space:]]*enable:[[:space:]]*)true/\1false/;p;}' ${CONFIG} &>/dev/null
-sed -i "s/estimate-time: true/estimate-time: false/g" ${CONFIG}
+tmpconfig=$(mktemp ${LLSF_REFBOX_DIR}/cfg/store_to_report_generated_XXXXXX.yaml)
+
+TRAP_SIGNALS="SIGINT SIGTERM SIGPIPE EXIT"
+cleanup () {
+  trap - $TRAP_SIGNALS
+  rm -f $tmpconfig
+	killall -9 llsf-refbox &>/dev/null &
+}
+trap cleanup $TRAP_SIGNALS
+
 for i in  $(seq 1 $NUM_BENCHMARKS)
 do
 	DOC_COUNT=$(mongo rcll --eval "db.game_report.count({\"report-name\": \"${NAME_PREFIX}_${i}\"})" --quiet)
+
 	# make sure that the report name is fresh
 	if [ ${DOC_COUNT} -ne 0 ]; then
-    echo "Skipping ${NAME_PREFIX}_${i} as a report already exists"
+		echo "Skipping ${NAME_PREFIX}_${i} as a report already exists"
 	else
+		truncate -s 0 ${tmpconfig}
+		echo "llsfrb/game/store-to-report: \"${NAME_PREFIX}_${i}\"" > ${tmpconfig}
 		# in rare occasions the generation fails, hence simply retry when it happens
 		while [ ${DOC_COUNT} -ne 1 ]
 		do
-			sed -i "s/store-to-report: \".*\"/store-to-report: \"${NAME_PREFIX}_${i}\"/g" ${CONFIG}
-			${LLSF_REFBOX_DIR}/bin/./llsf-refbox &>/dev/null &
+			echo "attempt to generate report"
+			${LLSF_REFBOX_DIR}/bin/./llsf-refbox --cfg-mongodb mongodb/enable_mongodb.yaml \
+			                                     --cfg-mps mps/mockup_mps.yaml \
+			                                     --cfg-custom ${tmpconfig} \
+			                                     --dump-cfg &>/dev/null &
 			sleep 1
 			${LLSF_REFBOX_DIR}/bin/./rcll-refbox-instruct -p PRE_GAME -s RUNNING -c Carologistics &>/dev/null
 			sleep 2
-			${LLSF_REFBOX_DIR}/bin/./rcll-refbox-instruct -p SETUP -s RUNNING -c Carologistics &>/dev/null
-			sleep 2
-			${LLSF_REFBOX_DIR}/bin/./rcll-refbox-instruct -p PRODUCTION  &>/dev/null
-			sleep 2
+			${LLSF_REFBOX_DIR}/bin/./rcll-refbox-instruct -p PRODUCTION &>/dev/null
```

Comment:
Why no more setup phase?

Reply:
good question, it turns out that this leads to the refbox getting completely stuck. I had no time to see why this is the case... The refbox actually is already in the SETUP phase as the switch to running and with a set team will lead to setup phase anyways.